lib/sdl_x11.hpp

```diff
@@ -69,35 +71,67 @@ class SdlX11Platform final : public SdlNativePlatform
       SDL_UnloadObject(lib);
 #endif
 #endif // SDL_VIDEO_DRIVER_X11_XINPUT2
+
+      // Create pipe for external events
+      if (pipe(event_pfd) == -1)
+      {
+         perror("pipe()");
+         exit(EXIT_FAILURE);
+      }
+      int flag = fcntl(event_pfd[0], F_GETFL);
+      fcntl(event_pfd[0], F_SETFL, flag | O_NONBLOCK);
    }
+
+   ~SdlX11Platform()
+   {
+      close(event_pfd[0]);
+      close(event_pfd[1]);
+   }
+
    void WaitEvent()
    {
-      int nstr, nfd = 1;
+      int nstr, nfd = 2;
       struct pollfd pfd[2];
 
       pfd[0].fd     = ConnectionNumber(disp);
       pfd[0].events = POLLIN;
       pfd[0].revents = 0;
-      if (glvis_command && visualize == 1)
-      {
-         pfd[1].fd     = glvis_command->ReadFD();
-         pfd[1].events = POLLIN;
-         pfd[1].revents = 0;
-         nfd = 2;
-      }
+
+      pfd[1].fd     = event_pfd[0];
+      pfd[1].events = POLLIN;
+      pfd[1].revents = 0;
+
       do
       {
          nstr = poll(pfd, nfd, -1);
       }
       while (nstr == -1 && errno == EINTR);
 
       if (nstr == -1) { perror("poll()"); }
+
+      int n = 0;
+      // Read out the pending GLVisCommand-sent events, if any
+      do
+      {
+         char c[10];
+         n = read(event_pfd[0], c, 10);
+      }
+      while (n > 0);
+   }
+   void SendEvent()
+   {
+      char c = 's';
+      if (write(event_pfd[1], &c, 1) != 1)
```

Comment:
(minor) maybe we'll never use anything but a char here but replacing the `1`s with `sizeof(c)` might be better?